src/artifacts.js

```diff
@@ -509,6 +509,16 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
       key: artifact.details.key,
     };
 
+    res.set('x-amz-meta-runid', runId);
+    res.set('x-amz-meta-taskid', taskId);
+    res.set('x-amz-meta-name', name);
+    res.set('x-amz-meta-content-sha256', artifact.details.contentSha256);
+    res.set('x-amz-meta-content-length', artifact.details.contentLength);
+    res.set('x-amz-meta-transfer-sha256', artifact.details.transferSha256);
```

Comment:
transfer is unimportant, this confuses me a lot...

I assume the only reason we store this in `artifact.details.transferSha256` is to support idempotency in `createArtifact` calls.

Reply:
No, the transfer is actually the only important one in this case.  Cloud-mirror, or any other redirecter is only dealing ever in transfer-bytes.  It'd be silly to have cloud-mirror gunzip the artifact so that it could immediately re-gzip the file.  It is also critical to ensure that the exact bytes which were download are uploaded into the other region.

This has nothing to do with idempotency on createArtifact.

Reply:
> Cloud-mirror, or any other redirecter is only dealing ever in transfer-bytes

Fair argument, that makes a lot of sense to me...

Can we please rename this `x-taskcluster-location-transfer-sha256`, as it's a header that indicates the `transfer-sha256` of the `location` header, and this is a taskcluster specific concept.